prov/verbs/src/verbs_info.c

```diff
@@ -885,35 +887,75 @@ int fi_ibv_get_srcaddr_devs(struct fi_info *info)
 		return 0;
 	}
 
-	for (fi = info; fi; fi = fi->next) {
+	for (fi = *info; fi; fi = fi->next) {
+		dev_found = 0;
 		dlist_foreach_container(&verbs_devs, dev, entry)
-			if (!strncmp(fi->domain_attr->name, dev->name, strlen(dev->name)))
+			if (!strncmp(fi->domain_attr->name, dev->name, strlen(dev->name))) {
+				dev_found = 1;
 				break;
+			}
 
-		dlist_foreach_container(&dev->addrs, addr, entry) {
-			/* When a device has multiple interfaces/addresses configured
-			 * duplicate fi_info and add the address info. fi->src_addr
-			 * would have been set in the previous iteration */
-			if (fi->src_addr) {
-				if (!(add_info = fi_dupinfo(fi))) {
-					ret = -FI_ENOMEM;
-					goto out;
+		if (dev_found) {
```

Comment:
The branch and the dev_found variable can be dropped and this part of the code reverted to previous state.